tests/dara_collectors/test_sequential_data_collector.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+import torch
+from pytest_mock import MockerFixture
+from torch.utils.data import TensorDataset
+
+from src.data_collectors.data_collector import DataCollector
+from src.data_collectors.dynamics_data_collector import DynamicsDataCollector
+from src.data_collectors.sequential_data_collector import SequentialDataCollector as cls
+from src.utils.step_record import RecordKeys as RK
+
+
+class DataCollectorImple(DataCollector):
+    def collect(self, step_record):
+        return None
+
+    def get_data(self):
+        return None
+
+
+class TestSequentialDataCollector:
+    num_data_collectors = 3
+
+    @pytest.fixture
+    def data_collectors(self, mocker: MockerFixture):
+        data_collectors = []
+        for _ in range(self.num_data_collectors):
+            mock = mocker.Mock(spec=DataCollectorImple)
+            data_collectors.append(mock)
+        return data_collectors
```

Comment:
Mockを使う場合はDataCollectorクラスを直接モックしてOKです。
```suggestion
class TestSequentialDataCollector:
    num_data_collectors = 3

    @pytest.fixture
    def data_collectors(self, mocker: MockerFixture):
        data_collectors = []
        for _ in range(self.num_data_collectors):
            mock = mocker.Mock(spec=DataCollector)
            data_collectors.append(mock)
        return data_collectors
```